cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -346,8 +349,11 @@ public void updateApp(HttpRequest request, HttpResponder responder,
       throw new BadRequestException("Request body is invalid json: " + e.getMessage());
     }
 
+    boolean updateSchedules =
+      appRequest.canUpdateSchedules() == null ? appUpdateSchedules : appRequest.canUpdateSchedules();
+
     try {
-      applicationLifecycleService.updateApp(appId, appRequest, createProgramTerminator());
+      applicationLifecycleService.updateApp(appId, appRequest, createProgramTerminator(), updateSchedules);
```

Comment:
thing is deployApp method call in AppLifecycleService expects `updateSchedules`, this deploy app is called from AppLifecycleHttpHandler, so Handler anyway has that default value of update schedules, in order to not pass this update schedule here, there is duplication of getting the default update schedule behavior in handler and lifecycle service, which doesn't provide much advantage. i prefer to pass this value along.